test/it/parser/src/main/resources/case/ddl/create-table.xml

```diff
@@ -17,6 +17,12 @@
   -->
 
 <sql-parser-test-cases>
+    <create-table sql-case-id="create_table_with_backtick">
+        <table name="t_order" start-delimiter="`" end-delimiter="`" start-index="13" stop-index="23" />
```

Comment:
Hi @zihaoAK47 
In #26035, We know that the expected table name should be `t_order, but now the result of the test is t_order, is this not as expected?


The following is my test in native mysql:
<img width="561" alt="image" src="https://github.com/apache/shardingsphere/assets/5668787/b08a3ff6-30a4-4790-b96c-554db62b221c">

Reply:
@RaigorJiang Sorry, I misunderstood the purpose of the start-delimiter and end-delimiter attributes for processing backticks. I have now created a new method to retrieve the correct table name enclosed with backticks. Please check it, mentor.

Reply:
Hi @zihaoAK47 
There is a method `getQuotedContent` in `IdentifierValue`, which is used to obtain the complete content in the quotation marks. Do you think this method can meet the requirements?

```java
public static String getQuotedContent(final String text) {
        if (Strings.isNullOrEmpty(text)) {
            return text;
        }
        QuoteCharacter quoteCharacter = QuoteCharacter.getQuoteCharacter(text);
        if (QuoteCharacter.NONE == quoteCharacter) {
            return text.trim();
        }
        return text.substring(1, text.length() - 1);
    }
```

Reply:
@RaigorJiang Tutor, I think the IdentifierValue.getQuotedContent(final String text) method is sufficient if we only need to retrieve the content that is enclosed by special characters, but this method cannot obtain the actual content, such as the real table name of \`t_order\` , which is t_order.
\`\`\`t_order\`  real table name is  \`t_order

I am confused about whether the IdentifierValue property in the TableNameSegment class should represent the real table name. After processing with IdentifierValue.getQuotedContent, \`\`\`t_orde\` is transformed into \`\`t_order. However, I believe that the IdentifierValue property should represent the real table name, which should be \`t_order instead of \`\`t_order.